### PR TITLE
BREAKING CHANGE: Adl object refactor

### PR DESCRIPTION
**Pull request by ro-joowan**
This PR is mapped to this Swagger change: https://github.com/Azure/azure-rest-api-specs/pull/1429

* Revised the inheritance structure and broke up objects for customer-experience clarity

* For more details, please refer to the PR above

* Found two bugs that I needed to fix in order to run tests in Playback mode:

  * including vcrpy in requirements.txt
  * resource-manager to data-plane in swagger_to_sdk_config.json for catalog.json and job.json

* There were also a couple modifications that I made to azure-mgmt/tests/test_mgmt_datalake_analytics.py in order for the tests to pass.  However, all the tests in the automatic PR passed, which was odd.

### CODE REVIEW

**Comment by begoldsm** [comment]
@lmazuel here is the aggregated PR for python that I promised you :). I reviewed this with @ro-joowan and helped him get it done. For some reason the automatic PR passed while this one required some test updates, so I am glad that we did it this way and caught the test failures :).

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@ __pycache__/
 
 # Virtual environment
 env*/
+myEnv/
```
Comment (lmazuel) [inline .gitignore]: Please remove this :)

File: azure-mgmt-datalake-store/HISTORY.rst [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 Release History
 ===============
+0.2.0 (2017-07-25)
+++++++++++++++++++
+* Create an inheritance structure for GET and LIST ADLS accounts.
```
Comment (lmazuel) [inline azure-mgmt-datalake-store/HISTORY.rst]: ChangeLog should be customer focus: what is the actual difference for my code. If you change 90% of the code, but it's changing nothing, I don't want a ChangeLog. Opposite, if you change only  a comma, but it's breaking I want why.

Then, the current bullet list is more a tasks list than a ChangeLog :)

File: requirements.txt [hunk 3]
```diff
@@ -2,3 +2,4 @@ futures;python_version<="2.7"
 python-dateutil
 pyopenssl
 msrestazure>=0.4.0,<0.5.0
+vcrpy
```
Comment (lmazuel) [inline requirements.txt]: Remove that, vcrpy is installed by the test requirements file (see .travis.yaml)

File: azure.pyproj [hunk 4]
```diff
@@ -2885,6 +2884,17 @@
     <Content Include="setup.cfg" />
     <Content Include="swagger_to_sdk_config.json" />
   </ItemGroup>
+  <ItemGroup>
```
Comment (lmazuel) [inline azure.pyproj]: I don't want this file in the PR, this is not supposed to be checked in

File: azure-mgmt-datalake-analytics/HISTORY.rst [hunk 5]
```diff
@@ -2,6 +2,22 @@
...
+    * For the properties use the right object (e.g. USqlJobProperties to CreateUSqlJobProperties)
+  * For building jobs, change JobInformation to BuildJobParameters
+    * For the properties use the right object (e.g. USqlJobProperties to CreateUSqlJobProperties)
+	* Note that the following fields are not part of BuildJobParameters
+	  * degreeOfParallelism
+	  * priority
+	  * related
+* Create a "Basic" jobInformation that is returned for LIST calls
+* Setup inheritance for GET jobs
+* Create an inheritance structure for GET and LIST ADLA accounts.
```
Comment (lmazuel) [inline azure-mgmt-datalake-analytics/HISTORY.rst]: See comments on the other ChangeLog

**Comment by ro-joowan** [comment]
Please merge and publish!

Thank you,
Joo Wan

**Pull request merged**